cpp/src/arrow/adapters/arrow-rados-cls/docs/architecture.md

```diff
@@ -0,0 +1,41 @@
+<!---
```

Comment:
This document is very good for helping to understand the intent of the PR but I'm not sure if we want this, as written, to be merged in.  I don't think your goal is to embed the Skyhook project inside of Arrow correct?  I think the goal is only to include the Arrow-Rados integration (and the Arrow Ceph object class)?  I may be misunderstanding things too.

Reply:
@westonpace Thank you very much for your review. Since the beginning of this year the Skyhook project _is_ the Arrow-RADOS integration. They have become the same because we discovered that Arrow had already all the parts we needed.

Reply:
@westonpace After discussing this with @JayjeetAtGithub, I confirm that the goal is to only include the Arrow-RADOS integration (including the Arrow Ceph object class). Jayjeet will update [the architecture document](https://github.com/uccross/skyhookdm-arrow/blob/rados-parquet-pr/cpp/src/arrow/adapters/arrow-rados-cls/docs/architecture.md) to highlight the parts that will be part of this pull request. The current plan is to put all other parts into a separate repository that will have Arrow as a submodule.